src/main/scala/com/fulcrumgenomics/bam/ClipBam.scala

```diff
@@ -112,36 +121,160 @@ class ClipBam
+    val numFivePrime  = this.clipper.clip5PrimeEndOfRead(frag, readOneFivePrime)
+    val numThreePrime = this.clipper.clip3PrimeEndOfRead(frag, readOneThreePrime)
+
+    // Update metrics
+    metric.foreach { m =>
+      m.update(
+        rec                 = frag,
+        priorBasesClipped   = priorBasesClipped,
+        numFivePrime        = numFivePrime,
+        numThreePrime       = numThreePrime,
+        numOverlappingBases = 0
+      )
+    }
+  }
+
+  /** Clips a fixed amount from the reads and then clips overlapping reads.
+    */
+  private[bam] def clipPair(r1: SamRecord, r2: SamRecord, r1Metric: Option[ClippingMetrics] = None, r2Metric: Option[ClippingMetrics] = None): Unit = {
+    val priorBasesClippedReadOne = r1.cigar.clippedBases
+    val priorBasesClippedReadTwo = r2.cigar.clippedBases
 
     // Clip the read!
-    this.clipper.clip5PrimeEndOfRead(r1, readOneFivePrime)
-    this.clipper.clip3PrimeEndOfRead(r1, readOneThreePrime)
-    this.clipper.clip5PrimeEndOfRead(r2, readTwoFivePrime)
-    this.clipper.clip3PrimeEndOfRead(r2, readTwoThreePrime)
-
-    if (clipOverlappingReads && r1.mapped && r2.mapped && r1.refIndex == r2.refIndex && r1.pairOrientation == PairOrientation.FR) {
-      val (f,r) = if (r1.negativeStrand) (r2, r1) else (r1, r2)
-      clipOverlappingReads(f, r)
+    val numReadOneFivePrime  = this.clipper.clip5PrimeEndOfRead(r1, readOneFivePrime)
+    val numReadOneThreePrime = this.clipper.clip3PrimeEndOfRead(r1, readOneThreePrime)
+    val numReadTwoFivePrime  = this.clipper.clip5PrimeEndOfRead(r2, readTwoFivePrime)
+    val numReadTwoThreePrime = this.clipper.clip3PrimeEndOfRead(r2, readTwoThreePrime)
+
+    val (numOverlappingBasesReadOne, numOverlappingBasesReadTwo) = {
+      if (clipOverlappingReads && r1.mapped && r2.mapped && r1.refIndex == r2.refIndex && r1.pairOrientation == PairOrientation.FR) {
+        val (f,r) = if (r1.negativeStrand) (r2, r1) else (r1, r2)
+        clipOverlappingReads(f, r)
+      } else (0, 0)
+    }
+
+    r1Metric.foreach { m =>
+      m.update(
+        rec                 = r1,
+        priorBasesClipped   = priorBasesClippedReadOne,
+        numFivePrime        = numReadOneFivePrime,
+        numThreePrime       = numReadOneThreePrime,
+        numOverlappingBases = numOverlappingBasesReadOne
+      )
+    }
+
+    r2Metric.foreach { m =>
+      m.update(
+        rec                 = r2,
+        priorBasesClipped   = priorBasesClippedReadTwo,
+        numFivePrime        = numReadTwoFivePrime,
+        numThreePrime       = numReadTwoThreePrime,
+        numOverlappingBases = numOverlappingBasesReadTwo
+      )
     }
   }
 
   /** Clips overlapping reads, where both ends of the read pair are mapped to the same chromosome, and in FR orientation. */
-  protected def clipOverlappingReads(f: SamRecord, r: SamRecord): Unit = {
+  protected def clipOverlappingReads(f: SamRecord, r: SamRecord): (Int, Int) = {
+    var numOverlappingBasesReadOne: Int = 0
```

Comment:
Technically these are more like `numClippedFromReadOne` and `numClippedFromReadTwo` right?  Since it can be more than the overlap in some cases.

Reply:
This method only clips the overlap, so it is exactly the bases clipped due to overlapping reads.